src/future/poll_fn.rs

```diff
@@ -0,0 +1,57 @@
+//! Definition of the `PollFn` adapter combinator
+
+use core::fmt;
+use core::pin::Pin;
+use std::future::Future;
+use std::task::{Context, Poll};
+
+/// Future for the [`poll_fn`] function.
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+struct PollFn<F> {
```

Comment:
So this struct is private, which means it's debug impl, must_use, and unpin impl do nothing (dead code).

What do you think would be the end goal for these structs? Should we eventually make them public or keep functions like `poll_fn` just `async fn`s?

Reply:
I think `poll_fn` should stay an `async fn`; it looks nice, and I don't see much value in having the actual struct be public.

Also fwiw `Unpin` *does* actually do something here for us right now, so I'll keep it (else it doesn't compile).